cluster/service/cluster.go

```diff
@@ -146,3 +152,71 @@ func (c clusterService) InitializeClusterWatcher() (func() error, error) {
 
 	return watcher.Close, err
 }
+
+// LinkIdentityToCluster links Identity to Cluster
+func (c clusterService) LinkIdentityToCluster(ctx context.Context, identityID, clusterURL string) error {
+	if err := validateParams(identityID, clusterURL); err != nil {
+		return errs.Wrapf(err, "failed to link identity %s to cluster with url '%s'", identityID, clusterURL)
+	}
+
+	id, err := uuid.FromString(identityID)
+	if err != nil {
+		return errors.NewBadParameterError("identity-id", fmt.Sprintf("identity-id %s is not a valid UUID", identityID))
+	}
+
+	rc, err := c.Repositories().Clusters().LoadClusterByURL(ctx, clusterURL)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"cluster_url": clusterURL,
+			"err":         err,
+		}, "failed to load cluster with url %s", clusterURL)
+		return errors.NewBadParameterError("cluster-url", fmt.Sprintf("cluster with requested url %s doesn't exist", clusterURL))
+	}
+
+	clusterID := rc.ClusterID
+	_, err = c.Repositories().IdentityClusters().Load(ctx, id, clusterID)
+	if err != nil {
+		if ok, _ := errors.IsNotFoundError(err); ok {
```

Comment:
Shouldn't we check ignore_if_exists param before trying to check if it's already exists?